### PR TITLE
Initialization with Job API [v2]

### PR DESCRIPTION
**Pull request by richtja**
Right now when we want to run tests through avocado Job API the avocado-vt is
not loaded, because a lot of avocado-vt settings is initialized through CLI
plugin. This commit moves the initialization to the Init plugin and leaves CLI
plugin just for necessary command line setup

Signed-off-by: Jan Richter jarichte@redhat.com

---
Changes form v1:

- put style fixes to the separated commits
- usage of `virttest.compat`

### CODE REVIEW

**Review by clebergnu — APPROVED** [review]
LGTM, tested and even built some patches on top of this (coming soon!).

Thanks!

**Comment by pevogam** [comment]
I think this PR is the cause for the issue I reported in https://github.com/avocado-framework/avocado-vt/issues/2912.

@richtja As much as I understand wishing to run unit tests through the Avocado API, I think having working settings has a higher priority. See the issue above for more details.

**Comment by pevogam** [comment]
@clebergnu I think PRs in Avocado VT need two approvals in order to get merged (as per community policy) and it seems this one only got one approval.

**Comment by richtja** [comment]
> I think this PR is the cause for the issue I reported in #2912.
> 
> @richtja As much as I understand wishing to run unit tests through the Avocado API, I think having working settings has a higher priority. See the issue above for more details.

@pevogam Thanks for letting me know, I will investigate that.

**Comment by richtja** [comment]
> @clebergnu I think PRs in Avocado VT need two approvals in order to get merged (as per community policy) and it seems this one only got one approval.

@pevogam Right now in GitHub setting s is set to one approval that's why I merged this after one approval. I agree with you that two approvals are needed, so we should change the GitHub setting to have it more clear. What do you think?

**Comment by pevogam** [comment]
> > @clebergnu I think PRs in Avocado VT need two approvals in order to get merged (as per community policy) and it seems this one only got one approval.
> 
> @pevogam Right now in GitHub setting s is set to one approval that's why I merged this after one approval. I agree with you that two approvals are needed, so we should change the GitHub setting to have it more clear. What do you think?

Yes, I agree especially because this is the rule that is already followed in Avocado VT. It is also a blocker for the PR-s for the rest of us so it is good if the rule applies to everyone and not just contributors without admin access.

**Pull request merged**
